### ISSUE TITLE
How can I disable the verdict print message?

### ISSUE THREAD

**Issue by pedrovagner** [netromdk/vermin#85]
Executing `vermin -q -t=3.7- some/file.py` prints:
```
Minimum required versions: 3.5
Incompatible versions:     2
```
I'm interested only on the returned code when succeed. Option `--quiet` is suppose to do that, isn't? How can I disable this message? This message may lead others to think something is wrong.

On my projects, besides `vermin`, I usually ran  `pyflakes` and `pycodestyle`, both prints nothing when succeed, or print violation messages when fail.

For now, I solved this question by redirecting stdout `vermin -q -t=3.7- some/file.py >/dev/null` but I completely loose the error messages.

**Comment by netromdk** [comment]
Thanks for the feedback.

That's a great question. It has always meant quiet but with the verdict lines only:
```
  --quiet | -q
        Quiet mode. It only prints the final versions verdict.
```

But if no tips are going to be shown (or `--no-tips` is used) then only the verdict will be shown.
There are a few other cases and important ones yield exit code 1 anyway.
So I might agree that it should be changed to be fully quiet, showing nothing.

I've pushed a commit to a branch I'll merge to `master`, I think, this weekend and then publish a new version.

**Comment by pedrovagner** [comment]
@netromdk thanks man, you are awesome.

I will soon put a public PGP key on my site (just like you did on yours). It is a difficult task to convince (to educate) my colleagues to exchange encrypted email messages. I usually said that mail protocols needs to be fixed, but tech giants don't want that.

Getting back to the issue.

I usually verify a lot of files, so the use of `--violations` is very useful, but it's incompatible with `--quiet`. Please, make it compatible.

Would be nice if `--violations` indicates the file line. I know, I'm asking to much. This is not important for now.

In short, `--quiet` should print nothing on success, but print normally on error.

**Comment by netromdk** [comment]
I wholeheartedly agree about enabling for encryption and signatures when sending emails (well, in all cases, really!).

The way `--violations` work is by internally setting verbosity to at least level 2 but it can be increased for more information.

If you want it to show the lines, you can just use verbosity level 3: `--violations -vvv`. It'll show like
```
2.7, 3.0     test.py
  L17: set comprehensions requires 2.7, 3.0
```
Instead of the following using only `--violations`:
```
2.7, 3.0     test.py
  set comprehensions requires 2.7, 3.0
```

Having both `--quiet` and verbosity thus doesn't make sense. I think I prefer the quiet mode being quiet.

Btw. you can also use the "parsable" output format instead of the default one. It works as you say, especially for violations mode because it _only_ shows rule findings.

For instance, a file might give the following results in parsable format:
```
% vermin -f parsable /path/to/examples/abc.py
/path/to/examples/abc.py:1:7:2.6:3.0:'abc' module
/path/to/examples/abc.py:2::!2:3.4:'abc.ABC' member
/path/to/examples/abc.py:::!2:3.4:
:::!2:3.4:
```
Each file shows its minimum required versions, and then the final verdict.

With violations and target 3.4, it only shows the final verdict because nothing violates it:
```
% vermin --violations -t=3.4 --format parsable examples/abc.py
:::!2:3.4:
% echo $?
0
```

Otherwise, maybe using `--violations` could make some changes to what is shown or a new argument could be added. But it still comes back to that it's hard to both have verbosity level 2+ _and_ being quiet at the same time.

What about this: even though `--violations` sets verbosity level at least 2 maybe everything before and after parsing and processing could function in "quiet mode". _Or_ maybe it allows to use `--violations --quiet` at the same time to function exactly like I just wrote. I think this is best because not all want it automatically all quiet.

**Comment by netromdk** [comment]
With that idea in mind, this is how it would work (I have a quick version running already).

## Case 0: no violations
```
% vermin --violations -t=3.4 examples/abc.py ; echo $?
Detecting python files..
Analyzing using 10 processes..
Tips:
- Since '# novm' or '# novermin' weren't used, a speedup can be achieved using: --no-parse-comments
(disable using: --no-tips)

Minimum required versions: 3.4
Incompatible versions:     2
0
```

## Case 1: no violations + quiet mode
```
% vermin --violations --quiet -t=3.4 examples/abc.py ; echo $?
0
```
This is what I think you would like to happen when there are no violations. Keep in mind that the "0" is from the `echo` in the end.

## Case 2: violations
```
% ./vermin.py --violations -t=3.3 examples/abc.py ; echo $?
Detecting python files..
Analyzing using 10 processes..
!2, 3.4      /Users/netrom/git/vermin/examples/abc.py
  'abc.ABC' member requires !2, 3.4

Tips:
- Since '# novm' or '# novermin' weren't used, a speedup can be achieved using: --no-parse-comments
(disable using: --no-tips)

Minimum required versions: 3.4
Incompatible versions:     2
Target versions not met:   3.3
1
```

## Case 3: violations + quiet mode
```
% vermin --violations --quiet -t=3.3 examples/abc.py ; echo $?
!2, 3.4      /Users/netrom/git/vermin/examples/abc.py
  'abc.ABC' member requires !2, 3.4

1
```
So if there is a violation, it only shows that violation if used together with quiet mode.

Sound good?

**Comment by netromdk** [comment]
I actually am going to use it for my own `self-test`:
```
self-test:
	./vermin.py --violations --quiet -t=2.7 -t=3 ${VERMIN_FILES}
```

**Comment by pedrovagner** [comment]
@netromdk I just did some tests on `issue-84` and everything seems to be working great! Thanks man.

When these improvements will be merged into master?

**Comment by netromdk** [comment]
Glad to hear it, @pedrovagner! I'm going to release 1.4.0 tonight (soon).
